enterprise/meta.go

```diff
@@ -404,6 +409,22 @@ func (d *defaultClient) Do(URL *url.URL, path, method string, params map[string]
 
 }
 
+// AuthedCheckRedirect tries to follow the Influx Enterprise pattern of
+// redirecting to the leader but preserving authentication headers.
+func AuthedCheckRedirect(req *http.Request, via []*http.Request) error {
+	if len(via) >= 10 {
+		return errors.New("too many redirects")
+	} else if len(via) == 0 {
+		return nil
+	}
+	for attr, val := range via[0].Header {
```

Comment:
I don't _think_ this is a problem, but is there any hazard in copying all of the headers as opposed to just the `Authentication` one?